runtime/sema/type.go

```diff
@@ -361,6 +361,46 @@ const toStringFunctionDocString = `
 A textual representation of this object
 `
 
+// fromString
+const FromStringFunctionName = "fromString"
+
+func FromStringMetadata(ty Type) (methodType *FunctionType, docString string) {
+	methodType = &FunctionType{
+		Parameters: []*Parameter{
+			{
+				Label:          ArgumentLabelNotRequired,
+				Identifier:     "input",
+				TypeAnnotation: NewTypeAnnotation(StringType),
+			},
+		},
+		ReturnTypeAnnotation: NewTypeAnnotation(
+			&OptionalType{ty},
+		),
+	}
```

Comment:
Maybe split the type creation and the doc-comment creation into two methods. Currently, it seems to do two independent tasks..